src/ApacheDestinyService.java

```diff
@@ -57,4 +90,55 @@ private String getContent(HttpResponse response) throws IOException {
         }
         return content.toString();
     }
+
+    //Is the below repetition bad and should it be public?
+    //whats the best way to test half way through without having to make the
+    //method public or modify all the code above to call it?
+    private String getSingleJsonField(String jsonString, String field) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode root = mapper.readTree(jsonString);
+
+        //Will only ever return the first found field
+        return root.findValue(field).asText();
+    }
+
+    // Returns a list of JsonNodes for the field specified, eg a list of characters or items.
+    //Note that sub-tree search ends if a field is found, so possible children of result nodes are not included.
+    private List<JsonNode> getListOfJsonField(String jsonString, String field) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode root = mapper.readTree(jsonString);
+
+        return root.findValues(field);
+    }
+
+    //Using string here because i know i can cast it after the user has input it, it will also allow the user to
+    //enter a non int/long etc number without throwing an error, whats the best way to handle this?
+    private String userInputSelectCharacter(List<JsonNode> listOfCharacters) {
+        String userInput;
+        List<String> stringListOfCharacters = new ArrayList<String>();
+
+        for (JsonNode node : listOfCharacters) {
+            stringListOfCharacters.add(node.asText());
+        }
+
+        System.out.println(stringListOfCharacters);
```

Comment:
As an extension, instead of making the user type the whole character ID, I guess we could list out the characters like below and let them choose by index e.g.

```
1. Hunter
2. Warlock
3. Titan
```